charts/mlp/values.yaml

```diff
@@ -0,0 +1,118 @@
+deployment:
+  image:
+    pullPolicy: IfNotPresent
+    registry: ghcr.io
+    repository: gojek/mlp
+    tag: v1.7.1
+  replicaCount: 1
+
+  # Additional labels to apply on the pod level
+  podLabels: {}
+
+  ## Configure resource requests and limits
+  ## ref: http://kubernetes.io/docs/user-guide/compute-resources/
+  ##
+  resources: {}
+  #   requests:
+  #     cpu: 100m
+  #     memory: 128Mi
+
+  livenessProbe:
+    path: "/v1/internal/live"
+  readinessProbe:
+    path: "/v1/internal/ready"
+
+  apiHost: "http://mlp/v1"
+  mlflowTrackingUrl: "http://mlflow.mlp"
+  environment: "production"
+
+  authorization:
+    enabled: false
+    serverUrl: http://mlp-authorization-keto
+
+  teams:
+    - marketing
+    - operation
+    - business
+  streams:
+    - promotion-marketing
+    - operation-strategy
+    - business-analyst
+  docs:
+    [
+      {
+        "label": "Merlin User Guide",
+        "href": "https://github.com/gojek/merlin/blob/main/docs/getting-started/README.md",
+      },
+      {
+        "label": "Turing User Guide",
+        "href": "https://github.com/gojek/turing",
+      },
+      {
+        "label": "Feast User Guide",
+        "href": "https://docs.feast.dev/user-guide/overview",
+      },
+    ]
+  ui:
+    feastCoreApi: "http://feast.dev/v1"
+    merlinApi: "http://merlin.dev/v1"
+    turingApi: "http://turing.dev/v1"
+    clockworkHomepage: "http://clockwork.dev"
+    feastHomepage: "http://feast.dev"
+    kubeflowHomepage: "http://kubeflow.org"
+    merlinHomepage: "http://merlin.dev"
+    turingHomepage: "http://turing.dev"
+
+ingress:
+  enabled: false
+  # class:
+  # host:
+  # path: "/"
+
+encryption:
+  # encryption.key must be specified using --set flag.
+  key: "example-key-here"
+
+service:
+  externalPort: 8080
+  internalPort: 8080
+
+# If you would like to use an external postgres database, you can connect using these credentials
+externalPostgresql:
+  enabled: false
+  username: mlp
+  database: mlp
+  password: password
```

Comment:
we have another PR using postgresql https://github.com/caraml-dev/helm-charts/pull/8
I think we should consolidate the approach we use for external DB. I see a few possible scenarios here:
1. We have external postgres and we want to set the password in values.yaml
in this case, we should create a new secret (in caraml? mlp? both?) only when a password is provided

2. We have external postgres and we create a secret file manually
in this case, we shouldn't create a secret and just reference the external secret name instead of release-specific

3. We create a postgres in caraml and the user provides a password in one of these ways ([globals](https://github.com/bitnami/charts/blob/master/bitnami/postgresql/values.yaml#L27-L36), [values](https://github.com/bitnami/charts/blob/master/bitnami/postgresql/values.yaml#L121-L156))
4. We create a postgres in caraml and the user provides a password using an external secret 
5. We create a postgres in caraml and the user doesn't provide a password
In all this cases we need to reference the secret created by postgres instance

6. we create postgres for each dependency
Here we should reference secrets created by each dependency

Reply:
In this MLP chart, I've added support for 1,5,6. 

> 1. We have external Postgres and we want to set the password in values.yaml

In this one, CaraML should support an external Postgres for sub-components if they chose to use one, In that case, the secret will be created in the corresponding component only. (That is how it is done at the moment). MLP chart creates [postgres-secret](https://github.com/caraml-dev/helm-charts/pull/7/files#diff-7e3c0b3ea7142b17fb9b5d25fa19c6fb0edbccf4b1ca69819f890f4712154075)
for when it needs to connect to external Postgres. 

> 5. We create a Postgres in CaraML and the user doesn't provide a password

This can be the default option, CaraML creates a Postgres(which internally creates a secret) that can be used in all of the components referencing the same secret. UserName and DB name referenced via `global` values from CaraML chart. 

> 6. we create Postgres for each dependency

If the user wishes to do so, They should be able to do it. 


Looking at the CaraML Store MR, It looks like they use an external Postgres providing values in an [application.yaml](https://source.golabs.io/data-science/mlp/feast/caraml-store-ci/-/blob/master/values/e2e-override.yaml) file that is added in values for that chart. 



In my opinion, We can start with this:
CaraML will create a Postgres for all components to use (approach 5 from above). If a component wishes to use an existing external database or use a component-specific database for some reason it can choose to do so by enabling the corresponding values.

Reply:
As discussed, We are going to provide the following:

1. CaraML postgres (default)
2. Component level Postgres (eg: caraml-mlp-postgresql/ caraml-merlin-postgresql)
3. Component level external DB
  a. password provided through --set flag during run time 
  b. password provided as an external secret